Wykonano zadanie

a-dreszler opened the pull request.

a-dreszler committed:
Wykonano zadanie

slawekludwiczak reviewed (changes requested):
Postaraj się uprościć ten kod, trochę niepotrzebnie masz to wszystko rozbite na tyle etapów, czyli: wczytanie pliku jako String, później zamiana tego na tablicę String[], później zamiana na tablicę Operation[] i dopiero wyliczanie wyników. Można by od razu przy wczytwaniu pliku odczytać to jako tablica Operation[] i kod będzie dużo prostszy.

code review https://javastartbootcamp.slack.com/archives/C04LP8YSVQX/p1682609887890609

a-dreszler committed:
Wprowadzono poprawki:
 > Od teraz program zamiast wczytywać całość pliku jako String i rozbijać go na poszczególne działania, czyta plik linijka po linijce i w każdym kroku zamienia linijkę z działaniem na obiekt Operation
   > W związku z tym od teraz metoda readOperations (wcześniej readFile) zwraca tablicę typu Operation[], którą podaje się jako parametr do metody printResults w klasie Calculator lub do metody saveToFile.
 > Wynik działania od teraz obliczany jest w konstruktorze klasy Calculator
 > W sytuacjach gdy zachodzi catch and re-throw, rzucany ponownie jest wyjątek niekontrolowany
 > Dodano weryfikację czy nie wprowadzono wartości null do metod z klasy FileUtils i Calculator

slawekludwiczak commented on src/main/java/FileUtils.java:
```diff
@@ -0,0 +1,104 @@
+    }
+
+    private static int countLines(String filePath) throws FileNotFoundException {
+        checkIfFilePathIsNull(filePath);
```
To jest metoda prywatna. Wywołujesz ją tylko wewnątrz readOperations, gdzie już sprawdziłeś, czy ścieżka do pliku nie jest nullem. Tutaj nie ma potrzeby sprawdzać tego ponownie. Ogólnie w wewnętrznym / prywatnym api można zazwyczaj założyć, że przekazany argument jest poprawny, bo to Ty decydujesz co będzie tutaj przekazane. W przypadku metod publicznych trzeba założyć, że ktoś inny może wywołać taką metodę z dowolnym argumentem i wtedy dodatkowe zabezpieczenia mają sens

slawekludwiczak commented on src/main/java/FileUtils.java:
```diff
@@ -0,0 +1,104 @@
+    }
+
+    private static void createFile(String fileName) {
+        checkIfFilePathIsNull(fileName);
```
podobna uwaga jak w countLines()

a-dreszler committed:
Wprowadzono poprawki:
 > Usunięto niepotrzebne null-checki
 > Od teraz w main łapane są tylko te wyjątki które są rzucane w kodzie wywoływanych metod
